crates/sncast/src/lib.rs

```diff
@@ -315,18 +315,78 @@ pub fn get_block_id(value: &str) -> Result<BlockId> {
     }
 }
 
+#[derive(Debug)]
+pub struct ErrorData {
+    pub data: String,
+}
+
+impl ErrorData {
+    #[must_use]
+    pub fn new(data: String) -> Self {
+        ErrorData { data }
+    }
+}
+
+impl From<ContractErrorData> for ErrorData {
+    fn from(value: ContractErrorData) -> Self {
+        ErrorData {
+            data: value.revert_error,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum TransactionError {
+    Rejected,
+    Reverted(ErrorData),
+}
+
+#[derive(Debug)]
+pub enum WaitForTransactionError {
+    TransactionError(TransactionError),
+    TimedOut,
+    Other(Error),
+}
+
+impl From<Error> for WaitForTransactionError {
+    fn from(value: Error) -> Self {
+        WaitForTransactionError::Other(value)
+    }
+}
+
+impl From<WaitForTransactionError> for Error {
+    fn from(value: WaitForTransactionError) -> Self {
+        match value {
+            WaitForTransactionError::Other(error) => error,
+            WaitForTransactionError::TimedOut => {
+                anyhow!("sncast timed out while waiting for transaction to succeed")
+            }
+            WaitForTransactionError::TransactionError(error) => match error {
+                TransactionError::Rejected => {
+                    anyhow!("Transaction has been rejected")
+                }
+                TransactionError::Reverted(ErrorData { data: reason }) => {
+                    anyhow!("Transaction has been reverted = {reason}")
+                }
+            },
+        }
+    }
+}
```

Comment:
this is very similar to impl from response/errors.rs right? is there a plan to merge them together, to avoid unnecessary duplications, or is it not easily-doable because of additional types there?

Reply:
Reworked it in https://github.com/foundry-rs/starknet-foundry/pull/1710/commits/d7de424a38c29328c6a0908163813f74f753174b